ledger/src/block/block_header.rs

```diff
@@ -59,6 +59,16 @@ impl BlockHeader {
         HEADER_SIZE
     }
 
+    /// Returns `true` if the block header is uniquely a genesis block header.
+    pub fn is_genesis(&self) -> bool {
+        // Ensure the previous block hash in the genesis block is 0.
+        self.previous_block_hash == BlockHeaderHash([0u8; 32])
+            // Ensure the timestamp in the genesis block is 0.
+            && self.time == 0
+            // Ensure the difficulty target in the genesis block is 0xFFFF_FFFF_FFFF_FFFF_u64.
+            && self.difficulty_target == 0xFFFF_FFFF_FFFF_FFFF_u64
```

Comment:
isn't any of these conditions sufficient to determine that something is a genesis block? if that's the case, this should be:
```
self.time == 0 || self.difficulty_target == 0xFFFF_FFFF_FFFF_FFFF_u64 || self.previous_block_hash == BlockHeaderHash([0u8; 32])
```
instead (with this altered order for faster short-circuiting)

Reply:
Thats a great point, will definitely reorder for faster short circuiting.

The previous block hash and time options can definitely be an `||`. The odds of hitting an all 0's block hash would be astonishing.

Now the difficulty target, I'm not too sure about. If the network drops in mining power, technically the min difficulty would be hit again (0xFFFF...).

Reply:
For now, I will remove the difficulty requirement here since it isn't as relevant.